lib/clients/BaseClient.py

```diff
@@ -143,11 +150,15 @@ def __getattribute__(self, attr):
         else:
             return method
 
-    def __retry(self, function, args):
+    def __retry(self, function, args, throw_exception=None):
         try:
             return self.__retry_rescuer(function, args)
         except Exception as error:
-            return None
+            if throw_exception == True:
```

Comment:
so services wont retry if upload or download fails ?

Reply:
AFAIK, this will retry for X number of times and then fail. While failing, this exception will be caught and thrown.

Reply:
ok , looks fine .. since retrying is done in rescuer function,